windows-agent/internal/proservices/landscape/utils.go

```diff
@@ -202,7 +203,12 @@ func (e newInstanceInfoMinorError) Error() string {
 func newInstanceInfo(d *distro.Distro) (info *landscapeapi.HostAgentInfo_InstanceInfo, err error) {
 	state, err := d.State()
```

Comment:
I would rather have this in a loop, with some timeouts, as we are uncertain and this could happen on startup, as you said, we are unsure when WSL would be "ready".

Reply:
Yeap! That makes sense. I saw this happening recently while the everything was up and running for a while.